presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java

```diff
@@ -214,7 +218,7 @@ private void asyncQueryResults(
     {
         Duration wait = WAIT_ORDERING.min(MAX_WAIT_TIME, maxWait);
         if (targetResultSize == null) {
```

Comment:
Can we wrap `targetResultSize` on top of `Optional` so that we can replace the null checks ?
It would be something like 
`targetResultSize.map(size -> Ordering.natural().min(size, MAX_TARGET_RESULT_SIZE).orElse(serverTargetResultSize)`

@kokosing  Your insights ?

Reply:
Yes, I suggested that in https://github.com/prestosql/presto/pull/3017/files#r390148921, after that `targetResultSize` would become `Optional` here

Reply:
I will send a PR to incorporate this and other suggestions here.